flows/knack/knack_banner.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+
+"""
+Name: Knack Banner HR App integration
+Description: Update knack HR app based on records in Banner and CTM
+Schedule: "45 13 * * *"
+Labels: atd-data02, knack
+"""
+
+import os
+import docker
+import prefect
+from datetime import datetime, timedelta
+
+# Prefect
+from prefect import Flow, task
+from prefect.engine.results import PrefectResult
+from prefect.tasks.prefect import create_flow_run, get_task_run_result
+from prefect.storage import GitHub
+from prefect.run_configs import UniversalRun
+from prefect.engine.state import Failed, TriggerFailed, Retrying
+from prefect.schedules import Schedule
+from prefect.schedules.clocks import CronClock
+from prefect.backend import get_key_value
+from prefect.triggers import all_successful
+
+from prefect.utilities.notifications import slack_notifier
+from prefect.tasks.notifications.email_task import EmailTask
+
+current_environment = os.getenv("PREFECT_CURRENT_ENVIRONMENT", "production")
+
+# Set up slack fail handler
+handler = slack_notifier(only_states=[Failed, TriggerFailed, Retrying])
+
+docker_image = f"atddocker/atd-knack-banner:{current_environment}"
+
+# Retrieve the email configuration
+email_config = get_key_value(key="aws_email_config")
+environment_variables = get_key_value(key=f"atd_knack_banner_{current_environment}")
+
+
+# Retrieve the provider's data
+@task(
+    name="HR knack banner integration",
+    max_retries=2,
+    retry_delay=timedelta(minutes=5),
+    state_handlers=[handler],
+    slug="knack-banner",
+)
+def knack_banner_update_employees():
+    response = (
+        docker.from_env()
+        .containers.run(
+            image=docker_image,
+            working_dir="/app",
+            command=f"./atd-knack-banner/update_employees.py",
+            environment=environment_variables,
+            volumes=None,
+            remove=True,
+            detach=False,
+            stdout=True,
+        )
+        .decode("utf-8")
+    )
+    logger = prefect.context.get("logger")
+    logger.info(response)
+    return response
+
+
+# Configure email task
+email_task = EmailTask(
+    name="email_task",
+    subject="Knack HR updates from Banner",
+    email_to=environment_variables["EMAIL_ADDRESS"],
+    email_to_cc="chia.berry@austintexas.gov",
```

Comment:
Do you want your email in this as it gets into `main`?

Reply:
good question. i keep going back and forth, ill put it in an environment variable.